include/rmm/mr/device/device_memory_resource.hpp

```diff
@@ -299,7 +299,7 @@ class device_memory_resource {
    *
    * @returns bool true if the resource supports non-null CUDA streams.
    */
-  [[nodiscard]] virtual bool supports_streams() const noexcept = 0;
+  [[nodiscard]] virtual bool supports_streams() const noexcept { return false; }
```

Comment:
**question (non-blocking)**: This remains virtual so that downstream users don't immediately break? (I must admit that inheritance in C++ is not one of my strong suits)

Reply:
Exactly. And it's being deprecated as soon as the two downstream PRs (already approved) merge. Then removed in the next release. Nobody uses this, and it's not part of `cuda::mr::async_memory_resource`